ocaml/typing/typecore.ml

```diff
@@ -3698,10 +3697,18 @@ and type_approx_aux env sexp in_function ty_expected =
   | None      -> match sexp.pexp_desc with
     Pexp_let (_, _, e) -> type_approx_aux env e None ty_expected
   | Pexp_fun (l, _, p, e) ->
+      let l, p = match p with
+      | {ppat_desc = Ppat_constraint (inner_pat,
+                      ({ ptyp_desc = Ptyp_extension ({txt = "src_pos"; _}, _); _} as ty)); _} ->
+          (* If the argument is a src_pos, translate the label using this
+             information. Otherwise, we don't care about the argument type *)
+          Typetexp.transl_label l (Some ty), inner_pat
```

Comment:
This call to `transl_label` duplicates the `src_pos` check above. Can you restructure the code so that the `src_pos` is checked in just one place?